crates/red_knot_project/src/walk.rs

```diff
@@ -129,7 +129,16 @@ impl<'a> ProjectFilesWalker<'a> {
     {
         let mut paths = paths.into_iter();
 
-        let mut walker = db.system().walk_directory(paths.next()?.as_ref());
+        let mut walker = db
+            .system()
+            .walk_directory(paths.next()?.as_ref())
+            .standard_filters(
+                db.project()
+                    .metadata(db)
+                    .options()
+                    .respect_ignore_files
+                    .unwrap_or(true),
```

Comment:
Red Knot has two structs related to configurations:

* `Options`: The options without the defaults filled in. Options can be merged from differnt sources. Only used during the start process.
* `Settings`: Has a very similar structure to `Options` but all defaults are filled in and the options were validated. Created by calling `Options::to_settings`. Used inside red knot when looking up a configuration value


That's why we should add `respect_ignore_files: bool` to the `Settings` struct, initialize the field in `Options::to_settings`, and use `project.settings().respect_ignore_files()` here.